sql/api/src/main/scala/org/apache/spark/sql/types/StringType.scala

```diff
@@ -61,6 +63,10 @@ class StringType private(val collationId: Int) extends AtomicType with Serializa
     if (isUTF8BinaryCollation) "string"
     else s"string collate ${CollationFactory.fetchCollation(collationId).collationName}"
 
+  // Due to backwards compatibility all string types are serialized in json as
```

Comment:
also mention compatibility with other readers and that the metadata is in StructField